client/state/selectors/test/site-supports-realtime-backup.js

```diff
@@ -0,0 +1,218 @@
+/**
+ * Internal dependencies
+ */
+import siteSupportsRealtimeBackup from 'state/selectors/site-supports-realtime-backup';
+import {
+	PRODUCT_JETPACK_BACKUP_DAILY,
+	PRODUCT_JETPACK_BACKUP_DAILY_MONTHLY,
+	PRODUCT_JETPACK_BACKUP_REALTIME,
+	PRODUCT_JETPACK_BACKUP_REALTIME_MONTHLY,
+} from 'lib/products-values/constants';
+
+describe( 'siteSupportsRealtimeBackup()', () => {
+	test( 'should return false when no data is available', () => {
+		const siteId = 123456;
+		const state = {
+			purchases: {
+				data: [],
+			},
+			sites: {
+				plans: {},
+			},
+		};
+
+		expect( siteSupportsRealtimeBackup( state, siteId ) ).toBe( false );
+	} );
+
+	test( 'should return false when there is an inactive real time backup purchase for that site', () => {
+		const siteId = 123456;
+		const state = {
+			purchases: {
+				data: [
+					{
+						active: false,
+						blog_id: siteId,
+						product_slug: PRODUCT_JETPACK_BACKUP_REALTIME,
+					},
+				],
+			},
+			sites: {
+				plans: {},
+			},
+		};
+
+		expect( siteSupportsRealtimeBackup( state, siteId ) ).toBe( false );
+	} );
+
+	test( 'should return false when there is a yearly daily backup purchase for that site', () => {
+		const siteId = 123456;
+		const state = {
+			purchases: {
+				data: [
+					{
+						active: false,
+						blog_id: siteId,
+						product_slug: PRODUCT_JETPACK_BACKUP_DAILY,
+					},
+				],
+			},
+			sites: {
+				plans: {},
+			},
+		};
+
+		expect( siteSupportsRealtimeBackup( state, siteId ) ).toBe( false );
+	} );
+
+	test( 'should return false when there is a monthly daily backup purchase for that site', () => {
+		const siteId = 123456;
+		const state = {
+			purchases: {
+				data: [
+					{
+						active: true,
+						blog_id: siteId,
+						product_slug: PRODUCT_JETPACK_BACKUP_DAILY_MONTHLY,
+					},
+				],
+			},
+			sites: {
+				plans: {},
+			},
+		};
+
+		expect( siteSupportsRealtimeBackup( state, siteId ) ).toBe( false );
+	} );
+
+	test( 'should return true when there is a yearly real time backup purchase for that site', () => {
+		const siteId = 123456;
+		const state = {
+			purchases: {
+				data: [
+					{
+						active: true,
+						blog_id: siteId,
+						product_slug: PRODUCT_JETPACK_BACKUP_REALTIME,
+					},
+				],
+			},
+			sites: {
+				plans: {},
+			},
+		};
+
+		expect( siteSupportsRealtimeBackup( state, siteId ) ).toBe( true );
+	} );
+
+	test( 'should return true when there is a monthly real time backup purchase for that site', () => {
+		const siteId = 123456;
+		const state = {
+			purchases: {
+				data: [
+					{
+						active: true,
+						blog_id: siteId,
+						product_slug: PRODUCT_JETPACK_BACKUP_REALTIME_MONTHLY,
+					},
+				],
+			},
+			sites: {
+				plans: {},
+			},
+		};
+
+		expect( siteSupportsRealtimeBackup( state, siteId ) ).toBe( true );
+	} );
+
+	test( 'should return false when site is on a Jetpack Premium yearly plan', () => {
```

Comment:
While I prefer how expressive the previous version of the tests was, I have no strong feelings, so I updated them to use loops, because you're right - it cuts the test code almost in half. Done in https://github.com/Automattic/wp-calypso/pull/38473/commits/39bc0efcce98c81f8fb28725d9700be3248c8de0